fetch.bs

```diff
@@ -4295,31 +4326,40 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
-       <var>revalidatingFlag</var>.
-
-       <li>
-        <p>If the <var>revalidatingFlag</var> is set and <var>httpRequest</var>'s
-        <a for=request>cache mode</a> is neither "<code>force-cache</code>" nor
-        "<code>only-if-cached</code>", then:
-
+       <li><p>If <a for=request>stale content mode</a> is "<code>allow</code>" and
+       <var>storedResponse</var> is a <a>stale response inside revalidate window</a>, then:
         <ol>
-         <li><p>If <var>storedResponse</var>'s <a for=response>header list</a>
-         <a for="header list">contains</a> `<code>ETag</code>`, then <a for="header list">append</a>
-         `<code>If-None-Match</code>` with its value to <var>httpRequest</var>'s
-         <a for=request>header list</a>.
-
-         <li><p>If <var>storedResponse</var>'s <a for=response>header list</a>
-         <a for="header list">contains</a> `<code>Last-Modified</code>`, then
-         <a for="header list">append</a> `<code>If-Modified-Since</code>` with its value to
-         <var>httpRequest</var>'s <a for=request>header list</a>.
+         <li><p>Set <var>response</var> to <var>storedResponse</var>.
+         <li><p>Set <var>response</var>'s <a for=response>cache state</a> to "<code>local</code>".
+         <li><p>In parallel perform <a for=main>main fetch</a> using <var>request</var> with
+          <ul class=brief>
+          <li><p><i>stale content mode</i> set to "<code>disallow</code>" and
+          <li><p><i>service-workers mode</i> set to "<code>none</code>".
+         </ul>
         </ol>
+       <li><p>Otherwise:
+        <p>If <var>storedResponse</var> is a <a>stale response</a>, then set the
+        <var>revalidatingFlag</var>.
+        <p>If the <var>revalidatingFlag</var> is set and <a for=request>cache mode</a>
+        is not one of "<code>force-cache</code>", "<code>only-if-cached</code>", then:
+
+         <ol>
+          <li><p>If <var>storedResponse</var>'s <a for=response>header list</a>
+          <a for="header list">contains</a> `<code>ETag</code>`, then <a for="header list">append</a>
+          `<code>If-None-Match</code>` with its value to <var>httpRequest</var>'s
+          <a for=request>header list</a>.
+
+          <li><p>If <var>storedResponse</var>'s <a for=response>header list</a>
+          <a for="header list">contains</a> `<code>Last-Modified</code>`, then
+          <a for="header list">append</a> `<code>If-Modified-Since</code>` with its value to
+          <var>httpRequest</var>'s <a for=request>header list</a>.
+         </ol>
 
         <p class=note>See also the
         "<a href=https://tools.ietf.org/html/rfc7234#section-4.3.4>Sending a Validation Request</a>"
         chapter of <cite>HTTP Caching</cite> [[!HTTP-CACHING]].
 
-       <li><p>Otherwise, set <var>response</var> to <var>storedResponse</var> and set
+        <li><p>Otherwise, set <var>response</var> to <var>storedResponse</var> and set
```

Comment:
It might be because of the messed-up if statement, but it kinda looks like a stale response could be used even if the stale content mode is "disallow".

Reply:
An <ol> is missing here. The last Otherwise is supposed to be inside the main Otherwise.